proto/streamlit/proto/Common.proto

```diff
@@ -42,6 +42,29 @@ message UInt32Array {
   repeated uint32 data = 1;
 }
 
+// TODO(vdonato): Finalize the next two proto types. We currently have enough
+// information here to support pure OS use cases, but we'll need to coordinate
+// with the SiS team to verify that we're passing enough file metadata back for
+// them.
+message FileURLsRequest {
+  string request_id = 1;
+  repeated string file_names = 2;
+  string session_id = 3;
+}
+
+message FileURLsResponse {
+  // TODO(vdonato): Maybe come up with a better name for this proto type.
+  message FileURLs {
+    string file_id = 1;
+    string upload_url = 2;
+    string delete_url = 3;
```

Comment:
Should we also add `retrieve_url` just for completeness?

Reply:
I think it's probably best to leave it off for now since I can't see a world where we'd ever have to re-fetch the file via URL from the client